### PR TITLE
[Debugger] Add evaluation support for Ballerina query expressions

### PR DESCRIPTION
**Pull request by NipunaRanasinghe**
## Purpose
This PR 
- introduces debugger expression evaluation support for Ballerina query expressions
- adds integration tests for related scenarios 
- fixes https://github.com/ballerina-platform/ballerina-lang/issues/31945

## Approach
Please refer to the simple diagram below, which represents the the basic query evaluation flow in the debug server.

Untitled Diagram

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [x] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [x] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Scheduler.java [hunk 1]
```diff
@@ -551,6 +539,23 @@ public ListenerRegistry getListenerRegistry() {
...
+        } catch (Throwable t) {
+            // Log and continue with default
+            getErrorStream().println("ballerina: error occurred in scheduler while reading system variable:" +
+                    RuntimeConstants.BALLERINA_MAX_POOL_SIZE_ENV_VAR + ", " + t.getMessage());
+        }
+        return poolSize;
+    }
+
+    private static PrintStream getErrorStream() {
+        return System.err;
```
Comment (warunalakshitha) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Scheduler.java:556]: Do we need sperate method for this? Can we directly use System.err

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Scheduler.java [hunk 2]
```diff
@@ -328,6 +316,7 @@ private void run() {
 
                 item = group.get();
 
+                Strand previousStrand = numThreads == 1 ?  strandHolder.get().strand : null;
```
Comment (warunalakshitha) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Scheduler.java:319]: Shall we move this out of while loop. Do we need if check here.
Strand previousStrand = strandHolder.get().strand

Comment (NipunaRanasinghe) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Scheduler.java]: Moved to a class variable and assigned in constructor with 5aa13c7

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Scheduler.java (hunk 1 above)
Comment (NipunaRanasinghe) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Scheduler.java]: Moved to a static field with 5aa13c7

**Review by praveennadarajah — APPROVED** [review]

**Review by warunalakshitha — COMMENTED** [review]
runtime changes LGTM

**Comment by warunalakshitha** [comment]
Code coverage is too low. Show we write some tests for debugger evaluation?

**Comment by NipunaRanasinghe** [comment]
@warunalakshitha Tests are already added. This is happening as the code coverage is not calculated on debugger integration tests as of now (https://github.com/ballerina-platform/ballerina-lang/issues/31352). We'll try to fix this asap.

**Comment by NipunaRanasinghe** [comment]
> Code coverage is too low. Should we write some tests for debugger evaluation?

@warunalakshitha Tests are already added. This is happening as the code coverage is not calculated on debugger integration tests as of now (https://github.com/ballerina-platform/ballerina-lang/issues/31352). We'll try to fix this asap.

**Pull request merged**